services/app-web/src/common-code/domain-models/submission2.ts

```diff
@@ -0,0 +1,49 @@
+import { RevisionType, Submission2Status, Submission2Type } from './Submission2Type'
+
+
+// submissionStatus computes the current status of the submission based on 
+// the submit/unlock info on its revisions.
+// These methods ASSUME that revisions are returned most-recent-first.
```

Comment:
I would like to see some basic unit tests here because now we have business logic going on in these functions. Also I wonder if we should add some safe guard here to throw an error if revisions are obviously not properly ordered ?

Reply:
I did include unit tests for this. I’m going back and forth on checking createdAt ordering ourselves. It’s not like we’re going to have tons of revisions, so effeciancy isn’t a super argument, but it does seem like we should be able to order them correctly if we want.

Reply:
I think we should at least check  first and last  revision submitted at dates  and throw an specific error  if things are wrong. personally don't think we should be reordering in this function though that seems like it could be a separate function that we could call optionally if that error type is thrown.